src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -334,18 +334,9 @@ private StructuredAnalyzerConfigOptions GetOptions(in AnalyzerConfigOptionsCache
             }
 
             var filePath = GetEffectiveFilePath(documentState);
-            if (filePath == null)
-            {
-                return StructuredAnalyzerConfigOptions.Empty;
-            }
-
-            var legacyDocumentOptionsProvider = services.GetService<ILegacyDocumentOptionsProvider>();
-            if (legacyDocumentOptionsProvider != null)
-            {
-                return StructuredAnalyzerConfigOptions.Create(legacyDocumentOptionsProvider.GetOptions(projectState.Id, filePath));
```

Comment:
i was working in this method, trying to understand the contract of what it returned (like if it would always return a cached vlaue once computed).  this was hte only codepath that always returned a new instance.  But it seems to only be for VS4Mac.  so we can reomve that api and this code here.